### PR TITLE
[BROWSEUI] SHEnumClassesOfCategories(): E_INVALIDARG is not unexpected. CORE-11711

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Follow-up to fdfef818ef5dcb4d0132da8fe4e91eb00415db24.

JIRA issue: [CORE-11711](https://jira.reactos.org/browse/CORE-11711)

Cc @HBelusca

### CODE REVIEW

**Comment by HBelusca** [comment]
What test does this fix?

File: dll/win32/browseui/shellbars/CSHEnumClassesOfCategories.cpp [hunk 1]
```diff
@@ -342,7 +342,12 @@ extern "C" HRESULT WINAPI SHEnumClassesOfCategories(ULONG cImplemented, CATID *p
 
     hr = ShellObjectCreatorInit<CSHEnumClassesOfCategories>(
             cImplemented, pImplemented, cRequired, pRequired, IID_PPV_ARG(IEnumGUID, out));
-    if (FAILED_UNEXPECTEDLY(hr))
+    if (FAILED(hr))
+    {
+        if (hr != E_INVALIDARG)
+            FAILED_UNEXPECTEDLY(hr);
```
Comment (HBelusca) [inline dll/win32/browseui/shellbars/CSHEnumClassesOfCategories.cpp]: This line alone does not make sense...
Unless...... you just want to have the DPRINT being output there?

Comment (SergeGautherie) [reply dll/win32/browseui/shellbars/CSHEnumClassesOfCategories.cpp]: Yes, the behavior is unchanged.

**Comment by HBelusca** [comment]
Passing invalid parameters to these functions is not expected, so the DPRINT is correct.
It can be viewed as "expected" only in the case where you are fuzzing the function with potentially invalid parameters, as is done by the current testcase.

**Comment by SergeGautherie** [comment]
> Passing invalid parameters to these functions is not expected, so the DPRINT is correct.

I guess it depends on the point of view:
passing invalid args is unexpected, nonetheless failing in that case is expected...

**Pull request closed without merging**